rpc-rs/src/chunkify.rs

```diff
@@ -140,24 +149,28 @@ impl ChunkEndpoint {
 
 pub(crate) fn chunkify_endpoint(
     domain: Option<String>,
+    nc: async_nats::Client,
```

Comment:
The main objection is that it violates the object hierarchy and fattens the object by adding a reference to another thing. Prior to the chunking implementation, it was unaware of any concept of lattices.

[Example: With the current api, a service could use an RpcClient to send and receive messages to or from multiple lattices, however that would only work if all the lattices used the same (or no) nats creds, so that counter-example is more theoretical than practical). 

I don't think there would be any performance change to caching the endpoint because it's lightweight and it's only ever used for large messages.

If we cached the ChunkEndpoint, either we'd add lattice_prefix as a parameter to the constructor and create the ChunkEndpoint for all RpcClients, or add a field like `Arc<Mutex<Option<ChunkEndpoint>>>`, and create it on-demand. (RpcClient is cloneable so anything that changes needs to be cloneable and have interior mutability). 

Initial impression is that this would be messier to cache the endpoint. Also, before refactoring, I'd want to put more thought into the relationships between lattices, nats connections, and chunking buckets.